pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,762 @@
+	t.Helper()
+
+	tc := &testContext{}
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	t.Cleanup(cancel)
+	tc.ctx = ctx
+
+	tc.client = fake.NewSimpleClientset()
+	reactor := createReactor(tc.client.Tracker())
+	tc.client.PrependReactor("*", "*", reactor)
+
+	informerFactory := informers.NewSharedInformerFactory(tc.client, 0)
+
+	opts := []runtime.Option{
+		runtime.WithClientSet(tc.client),
+		runtime.WithInformerFactory(informerFactory),
+	}
+	fh, err := runtime.NewFramework(nil, nil, tc.ctx.Done(), opts...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pl, err := New(nil, fh, feature.Features{EnableDynamicResourceAllocation: true})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tc.p = pl.(*dynamicResources)
+
+	// The tests use the API to create the objects because then reactors
+	// get triggered.
+	for _, claim := range claims {
+		_, err := tc.client.ResourceV1alpha1().ResourceClaims(claim.Namespace).Create(tc.ctx, claim, metav1.CreateOptions{})
+		require.NoError(t, err, "create resource claim")
+	}
+	for _, class := range classes {
+		_, err := tc.client.ResourceV1alpha1().ResourceClasses().Create(tc.ctx, class, metav1.CreateOptions{})
+		require.NoError(t, err, "create resource class")
+	}
+	for _, scheduling := range schedulings {
+		_, err := tc.client.ResourceV1alpha1().PodSchedulings(scheduling.Namespace).Create(tc.ctx, scheduling, metav1.CreateOptions{})
+		require.NoError(t, err, "create pod scheduling")
+	}
+
+	informerFactory.Start(tc.ctx.Done())
+	t.Cleanup(func() {
+		cancel()
+		informerFactory.Shutdown()
+	})
+
+	informerFactory.WaitForCacheSync(tc.ctx.Done())
+
+	for _, node := range nodes {
+		nodeInfo := framework.NewNodeInfo()
+		nodeInfo.SetNode(node)
+		tc.nodeInfos = append(tc.nodeInfos, nodeInfo)
+	}
+	tc.state = framework.NewCycleState()
+
+	return tc
+}
+
+// createReactor implements the logic required for the UID and ResourceVersion fields to work when using
+// the fake client. Add it with client.PrependReactor to your fake client.
+func createReactor(tracker cgotesting.ObjectTracker) func(action cgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+	var uidCounter int
+	var resourceVersionCounter int
+	var mutex sync.Mutex
+
+	return func(action cgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+		createAction, ok := action.(cgotesting.CreateAction)
+		if !ok {
+			return false, nil, nil
+		}
+		obj, ok := createAction.GetObject().(metav1.Object)
+		if !ok {
+			return false, nil, nil
+		}
+
+		mutex.Lock()
+		defer mutex.Unlock()
+		switch action.GetVerb() {
+		case "create":
+			if obj.GetUID() != "" {
+				return true, nil, errors.New("UID must not be set on create")
+			}
+			if obj.GetResourceVersion() != "" {
+				return true, nil, errors.New("ResourceVersion must not be set on create")
+			}
+			obj.SetUID(types.UID(fmt.Sprintf("UID-%d", uidCounter)))
+			uidCounter++
+			obj.SetResourceVersion(fmt.Sprintf("REV-%d", resourceVersionCounter))
```

Comment:
It makes the fake client behave a bit more like a real API server. Long term I want to write test cases that simulate conflicts, then it will be needed. Shall I remove it for now?

At the moment, the tests only need UID to be set because the controller uses that field to distinguish between "I must create the PodScheduling object" and "I must update it".